### PR TITLE
Adding applicable licensing requirements

### PR DESCRIPTION
**Pull request by DesmondSanctity**
##### SUMMARY
This change adds new applicable licensing requirements to the documentation.

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
[https://github.com/ansible/ansible/blob/devel/docs/docsite/rst/dev_guide/shared_snippets/licensing.txt](https://github.com/ansible/ansible/blob/devel/docs/docsite/rst/dev_guide/shared_snippets/licensing.txt#L9)

##### ADDITIONAL INFORMATION
This PR fixes this issue #70544

### CODE REVIEW

File: docs/docsite/rst/dev_guide/shared_snippets/licensing.txt [hunk 1]
```diff
@@ -7,3 +7,6 @@
       * All other ``module_utils`` must be licensed under BSD, so GPL-licensed third-party and Galaxy modules can use them.
       * If there's doubt about the appropriate license for a file in ``module_utils``, the Ansible Core Team will decide during an Ansible Core Community Meeting.
    * All other files shipped with Ansible, including all modules, must be licensed under the GPL license (GPLv3 or later).
+   * Existing license requirements still apply to content in ansible/ansible (ansible-base).
```
Comment (bcoca) [inline docs/docsite/rst/dev_guide/shared_snippets/licensing.txt]: ansible-core

Comment (DesmondSanctity) [reply docs/docsite/rst/dev_guide/shared_snippets/licensing.txt]: fixing it up asap

Comment (DesmondSanctity) [reply docs/docsite/rst/dev_guide/shared_snippets/licensing.txt]: Done with the fix

**Review by samccann — APPROVED** [review]

**Comment by samccann** [comment]
Thanks @DesmondSanctity  for the Ansible docs fix!

**Pull request merged**